algebra/tests/matrix/tools.h

```diff
@@ -39,6 +39,10 @@ extern int algebraTests_createAndFill(matrix_t *M, unsigned int rows, unsigned i
 extern int algebraTests_matrixCopy(matrix_t *src, matrix_t *des);
 
 
+/* Transposes matrix by rearranging its elements. Does not change M->transposed */
```

Comment:
```suggestion
Transposes matrix by rearranging its elements. M must be allocated with matrix_bufAlloc(). Does not change M->transposed
```

Or even better - copy the temporary matrix data to `M` data after rearranging, so that it would be safe for matrices with static data buffer.

Reply:
I have chosen second option - with copying buffor